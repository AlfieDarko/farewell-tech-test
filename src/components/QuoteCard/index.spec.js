import React from "react"
import { render, screen } from "@testing-library/react"
import QuotesCard from "./index"
import "@testing-library/jest-dom/extend-expect"

describe("QuotesCard Component", () => {
  const props = {
    quote: "Hi, Super Nintendo Chalmers!",
    character: "Ralph Wiggum",
    image:
      "https://cdn.glitch.com/3c3ffadc-3406-4440-bb95-d40ec8fcde72%2FRalphWiggum.png?1497567511523",
    characterDirection: "Left",
  }

  it("Should find QuotesCard by its data attribute", () => {
    render(<QuotesCard {...props} />)
    expect(screen.getByTestId("quotes-card-Ralph")).toBeInTheDocument()
  })

  it("Should find the correct elements", () => {
    render(<QuotesCard {...props} />)
    expect(screen.getByTestId("quotes-card-text-Ralph")).toHaveTextContent(
      "Hi, Super Nintendo Chalmers!"
    )
    expect(screen.getByTestId("quotes-card-image-Ralph")).toBeInTheDocument()
  })

  it("Should match the snapshot", () => {
    const { asFragment } = render(<QuotesCard {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
