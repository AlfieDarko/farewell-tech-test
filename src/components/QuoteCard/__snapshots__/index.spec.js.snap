// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotesCard Component Should match the snapshot 1`] = `
<DocumentFragment>
  <section
    class="sc-bdnylx lajIrS"
    data-testid="quotes-card-Ralph"
    tabindex="0"
  >
    <p
      class="sc-dlnjPT ghhryl"
      data-testid="quotes-card-text-Ralph"
    >
      Hi, Super Nintendo Chalmers!
    </p>
    <img
      alt="an illustration of The Simpsons character Ralph Wiggum"
      class="sc-gtssRu eZeSWe"
      data-testid="quotes-card-image-Ralph"
      src="https://cdn.glitch.com/3c3ffadc-3406-4440-bb95-d40ec8fcde72%2FRalphWiggum.png?1497567511523"
    />
  </section>
</DocumentFragment>
`;
